procedures/foreman/branch.md.erb

```diff
@@ -69,32 +71,56 @@
   * [ ] [foreman-selinux](https://github.com/theforeman/foreman-selinux)
   * [ ] [smart-proxy](https://github.com/theforeman/smart-proxy)
 * [ ] Branch foreman-packaging
-  * [ ] Create rpm/<%= release %> and update `packages/foreman/foreman-release/foreman.gpg`, `mock/*.cfg`, `package_manifest.yaml`, `rel-eng/{releasers.conf,tito.props}` and `repoclosure/*.conf`
-  * [ ] Create deb/<%= release %>
-  * [ ] Update `.github/PULL_REQUEST_TEMPLATE.md` (by adding `[ ] <%= release %>` to it) in `master`
+  * [ ] Create `rpm/<%= release %>`
+  * [ ] Create `deb/<%= release %>`
 * [ ] Bump versions to <%= develop %>-develop
   `echo <%= develop %>.0-develop > VERSION`
   * [ ] [foreman](https://github.com/theforeman/foreman)
   Also change package.json version field to <%= develop %>.0
   * [ ] [foreman-installer](https://github.com/theforeman/foreman-installer)
   * [ ] [foreman-selinux](https://github.com/theforeman/foreman-selinux)
   * [ ] [smart-proxy](https://github.com/theforeman/smart-proxy)
-* [ ] Update foreman-packaging:
-  * [ ] rpm/develop: Update `rel-eng/{releasers.conf,tito.props}`, `packages/foreman/foreman{,-{installer,proxy,release,selinux}}/*.spec`
+
+# Preparing build systems
+
+## Release engineer
+
+* [ ] Update foreman-packaging rpm/develop to ensure nightlies build:
+  * [ ] rpm/develop:
+    * [ ] Update the build tag: `sed -i 's/fm<%= release.tr('.', '_') %>/fm<%= develop.tr('.', '_') %>/g' rel-eng/{releasers.conf,tito.props}`
+    * [ ] Set to version `<%= develop %>.0`, reset release to `1` in `packages/foreman/foreman{,-{installer,proxy,release,selinux}}/*.spec` and create a changelog using `obal changelog --message '- Bump version to <%= develop %>-develop' foreman{,-{installer,proxy,release,selinux}}`
   * [ ] deb/develop: `scripts/changelog.rb -v <%= develop %>.0-1 -m "Bump changelog to <%= develop %>.0 to match VERSION" debian/*/*/changelog`
+* [ ] Prepare build systems for <%= release %> release:
+  * [ ] foreman-packaging's rpm/<%= release %>
+    * [ ] Update `packages/foreman/foreman-release/foreman.gpg`, `mock/*.cfg`, `package_manifest.yaml`, `rel-eng/{releasers.conf,tito.props}` and `repoclosure/*.conf`
+  * [ ] Create mash configuration on Koji:
+    ```
+    bundle exec ./tools.rb mash-scripts configs/foreman/<%= release %>.yaml
+    scp mash_scripts/foreman/<%= release %>.0/*.mash root@koji.katello.org:/etc/mash/
+    ```
+    * [ ] Create mash scripts on Koji:
+    ```
+    ssh root@koji.katello.org
+    cd /usr/local/bin
+    # create foreman-client-mash-split-<%= release %>.py  foreman-mash-split-<%= release %>.py  foreman-rails-mash-split-<%= release %>.py based on the unversioned files
+    ```
+  * [ ] Add new plugin tags to Koji plugins mash script (foreman-mash-split-plugins.py), remove old ones (keep three)
+  * [ ] Add the new release to the JJB job definitions in [foreman-infra](https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/)
```

Comment:
This can also be done pre-branching i think? the pipelines would not pass until branching, but once branched everything will be ready

Reply:
True, but I'd rather not have red jobs when we can avoid it. Accepting failing jobs is a bad thing IMHO :) This can be prepared as a PR before branching but I'd like to avoid changing the process too much in this PR. Let's keep this to (mostly) assigning roles on existing steps.